src/lib/telemetry/TelemetryReporter.ts

```diff
@@ -98,16 +106,37 @@ export class TelemetryReporter {
     if (errors.length > 0) {
       await this.createTelemetryReporterMetaError(errors)
     }
+
     await this.dataPendingLock.unlock()
   }
 
-  public reportErrors(errors: any[]) {
-    return this.telemetryClient.reportErrors(errors)
+  public async reportErrors(errors: any[]) {
+    try {
+      await this.telemetryClient.reportErrors(errors)
+    } catch (err) {
+      await this.dataPendingLock.lock()
+      await ensureDir(join(TelemetryReporter.PENDING_DATA_DIR, 'errors'))
+      await writeJson(join(TelemetryReporter.PENDING_DATA_DIR, 'errors', randomBytes(8).toString('hex')), errors)
+      await this.createTelemetryReporterMetaError(err)
+      await this.dataPendingLock.unlock()
+    }
+  }
+
+  public async reportMetrics(metrics: MetricReport[]) {
+    try {
+      await this.telemetryClient.reportMetrics(metrics)
+    } catch (err) {
+      await this.dataPendingLock.lock()
+      await ensureDir(join(TelemetryReporter.PENDING_DATA_DIR, 'metrics'))
+      await writeJson(join(TelemetryReporter.PENDING_DATA_DIR, 'metrics', randomBytes(8).toString('hex')), metrics)
+      await this.createTelemetryReporterMetaError(err)
+      await this.dataPendingLock.unlock()
```

Comment:
_super-nit_: For minimizing code duplicates this could be:
```ts
private async treatReportError(reportError: any, reportType: 'error', errors: ErrorReport[])
private async treatReportError(reportError: any, reportType: 'metric', metrics: MetricReport[])
private async treatReportError(reportError, reportType, metricsOrErrors) {
  await this.dataPendingLock.lock()
  await ensureDir(join(TelemetryReporter.PENDING_DATA_DIR, reportType))
  await writeJson(join(TelemetryReporter.PENDING_DATA_DIR, reportType, randomBytes(8).toString('hex')), metricsOrErrors)
  await this.createTelemetryReporterMetaError(reportError)
  await this.dataPendingLock.unlock()
}  
```